lgc/builder/ImageBuilder.cpp

```diff
@@ -752,13 +752,14 @@ Value *ImageBuilder::CreateImageSampleConvertYCbCr(Type *resultTy, unsigned dim,
   Value *pResult = nullptr;
   Value *ycbcrSamplerDesc = CreateExtractValue(convertingSamplerDesc, 0);
 
-  // Extract YCbCr meta data
-  const SamplerYCbCrConversionMetaData yCbCrMetaData = {
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(4)))->getZExtValue()),
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(5)))->getZExtValue()),
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(6)))->getZExtValue()),
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(7)))->getZExtValue()),
-  };
+  // Extract YCbCr meta data, which is the last 4 DWORDs of convertingSamplerDesc
+  const auto yCbCrMetaData(std::move([&]() -> auto {
+    SamplerYCbCrConversionMetaData metaData;
+    for (unsigned i = 0; i < 4; ++i)
+      *(&metaData.word0.u32All + i) = static_cast<unsigned>(
+          cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(4 + i)))->getZExtValue());
+    return metaData;
```

Comment:
Wow. I meant something simple like this:
```
auto getYCbCrMetaElement = [this, &ycbcrSamplerDesc](unsigned idx) {
  return static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(4)))->getZExtValue())
};
// ...
yCbCrMetaData.word0.u32All = getYCbCrMetaElement(4);
yCbCrMetaData.word0.u32All = getYCbCrMetaElement(5);
// ...
```

Reply:
Oh, okay :)